plugins/webp-uploads/hooks.php

```diff
@@ -77,7 +77,10 @@ function webp_uploads_create_sources_property( array $metadata, int $attachment_
 		$metadata['sources'] = array();
 	}
 
-	if ( empty( $metadata['sources'][ $mime_type ] ) ) {
+	if (
+		! isset( $metadata['sources'][ $mime_type ] ) ||
+		! is_array( $metadata['sources'][ $mime_type ] )
+	) {
```

Comment:
Based on @felixarntz's prior suggestion, I just realized this can be further simplified:

```suggestion
	if ( ! isset( $metadata['sources'][ $mime_type ]['file'] ) ) {
```

This will automatically verify that the array is present _and_ that it has the expected shape (a `file` key).